diff --git a/apk-tools-sys/build.rs b/apk-tools-sys/build.rs
index e60ec51..7137887 100644
--- a/apk-tools-sys/build.rs
+++ b/apk-tools-sys/build.rs
@@ -29,6 +29,7 @@ fn generate_bindings(include_paths: &Vec<&PathBuf>) {
     for path in include_paths {
         builder = builder.clang_arg(format!("-I{}", path.to_str().unwrap()));
     }
+    builder = builder.clang_arg("-DCRYPTO_USE_OPENSSL");
 
     let bindings = builder.generate().expect("Unable to generate bindings");
 
@@ -43,7 +44,8 @@ fn generate_bindings(include_paths: &Vec<&PathBuf>) {
     cc::Build::new()
         .file(out_dir.join("wrap_static_fns.c"))
         .includes(c_paths)
-        .compile("wrap_static_fns");
+        .define("CRYPTO_USE_OPENSSL", "1")
+	.compile("wrap_static_fns");
 
     println!("cargo:rerun-if-changed=src/wrapper.h");
 }
diff --git a/apk-tools-sys/src/wrapper.h b/apk-tools-sys/src/wrapper.h
index dfe3044..4fdceaa 100644
--- a/apk-tools-sys/src/wrapper.h
+++ b/apk-tools-sys/src/wrapper.h
@@ -2,6 +2,7 @@
 #include <apk/apk_atom.h>
 //#include <apk/apk_balloc.h>
 #include <apk/apk_blob.h>
+#include <apk/apk_crypto_openssl.h>
 #include <apk/apk_crypto.h>
 #include <apk/apk_context.h>
 #include <apk/apk_ctype.h>
