diff --git a/meson.build b/meson.build
index 7954d10..07c2956 100644
--- a/meson.build
+++ b/meson.build
@@ -49,7 +49,8 @@ cargo_build = custom_target(
 
 test(
   'cargo test --all',
-  args: [cargo.full_path(), 'test', '--all', '--target-dir', meson.current_build_dir()],
+  cargo,
+  args: ['test', '--all', '--target-dir', meson.current_build_dir()],
   timeout: 3000,
   env: [
     'CBUILD=' + build_machine.cpu_family(),
