diff --git a/core/Services/Calendar/Util/ECalComponent.vala b/core/Services/Calendar/Util/ECalComponent.vala
index ce9ca5a3..17fc0447 100644
--- a/core/Services/Calendar/Util/ECalComponent.vala
+++ b/core/Services/Calendar/Util/ECalComponent.vala
@@ -72,8 +72,8 @@ namespace Calendar.Util {
         }
 
         ICal.Timezone system_timezone = time_manager.system_timezone;
-        time_t start_unix = start_time.as_timet_with_zone (system_timezone);
-        time_t end_unix = end_time.as_timet_with_zone (system_timezone);
+        int64 start_unix = start_time.as_timet_with_zone (system_timezone);
+        int64 end_unix = end_time.as_timet_with_zone (system_timezone);
 
         /* If the selected date is inside the event */
         if (start_unix < selected_date_unix && selected_date_unix_next < end_unix) {
